addons/account/tests/test_reconciliation.py

```diff
@@ -454,6 +456,136 @@ def test_revert_payment_and_reconcile(self):
+            )
+            self.company.period_lock_date = None
+
+        # Testing exchange_move.date = today
+        for move_type in ('in_invoice', 'out_invoice'):
+            inv = create_invoice(move_type, '2016-06-01')
+            pay = create_payment(inv, '2020-10-01')  # today
+            exchange_move = unreconcile_payment_and_get_exchange_move(pay)
+            reverted_exchange_move = get_reverted_exchange_move(exchange_move)
+            # exchange_move.date = today => reverted_exchange_move.date is today
+            self.assertEqual(
+                reverted_exchange_move.date,
+                fields.Date.from_string('2020-10-01'),
+                'reversed exchange move date %s for move type %s should be 2020-10-01'
+                % (reverted_exchange_move.date, move_type),
+            )
+
+        # Testing making invoice and payment for past dates & with period_lock_date > unreconcile_date
+        for move_type in ('in_invoice', 'out_invoice'):
+            inv = create_invoice(move_type, '2016-01-01', '2016-02-01')
+            pay = create_payment(inv, '2017-03-01', '2017-04-01')
+            self.company.period_lock_date = '2017-05-02'
+            exchange_move = unreconcile_payment_and_get_exchange_move(pay, '2017-05-01')
+            reverted_exchange_move = get_reverted_exchange_move(exchange_move)
+            self.assertEqual(
+                reverted_exchange_move.date,
+                fields.Date.from_string('2017-05-03'),
+                'reversed exchange move date %s for move type %s should be 2020-05-03'
+                % (reverted_exchange_move.date, move_type),
+            )
+
+            move_reversal = self.env['account.move.reversal'].with_context(
+                active_model="account.move",
+                active_ids=inv.ids
+            ).create({
+                'date': fields.Date.from_string('2017-05-01'),
+                'reason': 'no reason',
+                'refund_method': 'refund',
+            })
+            res = move_reversal.reverse_moves()
+            reversal = self.env['account.move'].browse(res['res_id'])
+            # date in wizard overrides _get_reverse_date()
+            self.assertEqual(
+                reversal.date,
+                fields.Date.from_string('2017-05-01'),
+                'reversed exchange move date %s for move type %s should be 2020-05-01'
+                % (reverted_exchange_move.date, move_type),
+            )
+
+            with freeze_time('2017-05-01'):
+                # default date is _get_reverse_date()
+                reversed = inv._reverse_moves()
+                self.assertEqual(
+                    reversed.date,
+                    fields.Date.from_string('2017-05-03' if move_type == 'in_invoice' else '2017-05-01'),
+                    'reversed exchange move date %s for move type %s should be 2020-05-03'
+                    % (reverted_exchange_move.date, move_type),
```

Comment:
for all these string formatting, you can use `f"reversed exchange move date {reverted_exchange_move.date} for move type {move_type} should be 2020-05-03"` if you like

Reply:
made assert_date function